### PR TITLE
Adds bootstrap errors for impulse responses in VAR environment

### PR DESCRIPTION
**Pull request by evivanov**
- [ ] closes #xxxx
- [ ] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/tsa/vector_ar/irf.py [hunk 1]
```diff
@@ -491,10 +495,10 @@ def err_band_sz3(self, orth=False, svar=False, repl=1000, signif=0.05,
                 for i in range(neqs):
                     gamma[p,1:,i,j] = W[j,k[j],i*periods:(i+1)*periods] * irf_resim[p,1:,i,j]
                     if i == neqs-1:
-                        gamma[p,1:,i,j] = W[j,k[j],i*periods:] * irf_resim[p,1:,i,j]
+                        gamma[p,1:,i,j] = W[j,k[j],i*pentiles
+        indx = round(signif/2*repl)-1,round((1-signif/2)*reriods:] * irf_resim[p,1:,i,j]
 
-        gamma_sort = np.sort(gamma, axis=0) #sort to get quantiles
-        indx = round(signif/2*repl)-1,round((1-signif/2)*repl)-1
+        gamma_sort = np.sort(gamma, axis=0) #sort to get quapl)-1
```
Comment (jseabold) [inline statsmodels/tsa/vector_ar/irf.py:501]: This looks like an accident. Can you back this change out?

File: statsmodels/tsa/vector_ar/irf.py [hunk 2]
```diff
@@ -504,6 +508,74 @@ def err_band_sz3(self, orth=False, svar=False, repl=1000, signif=0.05,
                 upper[:,i,j] = irfs[:,i,j] + gamma_sort[indx[1],:,i,j]
 
         return lower, upper
+    
+    def err_band_bootstrap(self, orth=False, svar=False, repl=100, signif=0.05,
+                     seed=None):
```
[the remark this answers is not in the document]
Comment (jseabold) [reply statsmodels/tsa/vector_ar/irf.py:513]: This change might need to wait since we already use the current pattern elsewhere.

**Review by jseabold — CHANGES_REQUESTED** [review]
Overall this looks correct to me.

There's an accidental change that needs to be fixed, some style fixes, and a few nit-picky "optimizations."

File: statsmodels/tsa/vector_ar/irf.py [hunk 3]
```diff
@@ -504,6 +508,74 @@ def err_band_sz3(self, orth=False, svar=False, repl=1000, signif=0.05,
...
+        def resample(e):
+            """Resample data randomly with replacement."""
+            ind = np.random.choice(len(e), size=len(e))
+            return e[ind]
+    
+        def simulate(resid, Y, intercept, coefs, k_ar):
+            """
+            Simulate VAR model with given resuduals
+            """
+            e_hatb = resample(resid)
```
[the remark this answers is not in the document]
Comment (evivanov) [reply statsmodels/tsa/vector_ar/irf.py:546]: Unfortunately not
Residuals are not 1-dimensional and the choice function cannot work with that. Maybe there is another way but I am not aware of it

**Comment by evivanov** [comment]
> Overall this looks correct to me.
> 
> There's an accidental change that needs to be fixed, some style fixes, and a few nit-picky "optimizations."

@jseabold, thank you for the comments! I have resolved most of the issues and added new commit

**Pull request closed without merging**